### ISSUE TITLE
Allow using `$HOME/.minecraft` as working directory

### ISSUE THREAD

**Issue by JasonHK** [flathub/com.mojang.Minecraft#56]
Since Forge expects the `.minecraft` directory to be located inside `$HOME`, can you provide an option to use `$HOME/.minecraft` as the working directory?

**Comment by AsciiWolf** [comment]
Do Forge allows setting custom Minecraft directories? If so, try setting `$XDG_DATA_HOME/minecraft` as Minecraft directory.

**Comment by Garbulix** [comment]
@AsciiWolf - I am afraid I don't understand.
Forge lets you to set Minecraft directory, but given directory should be where Minecraft really is. With Flatpak version od MC, it's `$HOME/.var/app/com.mojang.Minecraft/.minecraft`.
Does installing script is able create some kind of link that leads to the Minecraft directory?
`$HOME/.minecraft -> $HOME/.var/app/com.mojang.Minecraft/.minecraft`

In the other hand, I am not really sure if that solution would be good. Maybe there should be an announcement about Minecraft directory, instead of making that kind of link. The problem will occur if someone uses (or used) tarballed version of Minecraft downloaded from Minecraft.net website, because that particular Minecraft creates `.minecraft` folder in home directory (there would be some kind of "link collision").

I am not sure if I described it well (because of my not perfect English).

**Comment by AsciiWolf** [comment]
Hmm, I am not sure if the symlink solution would be possible to do without using nasty hacks or breaking compatibility for old installations that we have to support.

@p1u3o What is your opinion?

**Comment by violetmage** [comment]
@AsciiWolf

@Garbulix wants to use `--filesystem="~/.minecraft:rw"` instead of `--persist="~/.minecraft"`

only is problem, migrating flatpak persists to ~/.minecraft

idea is to make using forge, optifine, etc. work same as if download minecraft from mojang

**Comment by AsciiWolf** [comment]
As noted in [this comment](https://github.com/flathub/com.mojang.Minecraft/pull/78#issuecomment-803659607), it is possible that we migrate from XDG dir back to a non-XDG `~/.minecraft` one (inside Flatpak), so rewriting it with `--filesystem="~/.minecraft:rw` probably will be possible in the future.

**Comment by nanonyme** [comment]
The migration path seems quite quirky. Please *thoroughly* test what happens in the migration if you have data both outside sandbox and inside.

**Comment by nanonyme** [comment]
Maybe the only wonky case really here is what happens if you end up in a situation where you might want to clobber directory on the host....

**Comment by saltedcoffii** [comment]
I've used the quick fix of:
```shell
ln -s ~/.var/app/com.mojang.minecraft/.minecraft ~/.minecraft
```
This allows `.minecraft` to be accessible at its normal location from outside the sandbox, allowing the OptiFine installer and other things to work.

**Comment by InetAddress** [comment]
> As noted in [this comment](https://github.com/flathub/com.mojang.Minecraft/pull/78#issuecomment-803659607), it is possible that we migrate from XDG dir back to a non-XDG `~/.minecraft` one (inside Flatpak), so rewriting it with `--filesystem="~/.minecraft:rw` probably will be possible in the future.

Any news about this?

**Comment by AsciiWolf** [comment]
Yes. We migrated back to a non-XDG dir a while ago. Feel free to test it.

**Comment by InetAddress** [comment]
> Yes. We migrated back to a non-XDG dir a while ago. Feel free to test it.

Sorry, I meant Flatpak Minecraft using the $HOME/.minecraft path by default. This would allow people with no experience to install Optifine, LabyMod, Forge, or even resource packs, as they (and the installers) are used to the default ~/.minecraft/ path, and the installers use a graphical path chooser, as @Garbulix said
> you must give Forge installer absolute path to your .minecraft install, which you select graphically

![imagen](https://user-images.githubusercontent.com/15693124/148100702-38c332bf-a759-4e6e-8917-ef0a610677c7.png)
![imagen](https://user-images.githubusercontent.com/15693124/148101111-34e70545-e117-4f9a-a513-24a9b00903a1.png)

**Comment by vchernin** [comment]
FYI I have a local branch + flatpak extension that basically just makes the fabric loader installer run on the first launch. Then you can install the fabric loader + a modded launcher profile and so on.

TBH I'm not very happy with it though. It's not very easy to communicate that a Flatpak extension needs to be installed. It's hard to communicate there are extra steps if you want the fabric loader installer to show up again. 

It would be preferable to have mod support without special tricks from the user's perspective.

**Comment by InetAddress** [comment]
> FYI I have a local branch + flatpak extension that basically just makes the fabric loader installer run on the first launch. Then you can install the fabric loader + a modded launcher profile and so on.

Weird solution xd

> It's not very easy to communicate that a Flatpak extension needs to be installed to have easy fabric support. It's hard to communicate there are extra steps if you want the fabric loader installer to show up again. The same would be true for other mods and mod installers.
> It would be preferable to have mod support without special tricks from the user's perspective.

That's the point. Flatpaks are there to make deploying and getting apps easier, for both developers and new users. If Mojang owned this repository, they would manage to get the app working as if it was the tarball version.
